code/modules/mob/living/carbon/life.dm

```diff
@@ -518,6 +520,13 @@ All effects don't start immediately, but rather get worse over time; the rate is
 			clear_alert("drunk")
 			sound_environment_override = SOUND_ENVIRONMENT_NONE
 
+		if(HAS_TRAIT_FROM(src, TRAIT_CLUMSY, DRUNK_TRAIT))
+			if(drunkenness < 30)
+				REMOVE_TRAIT(src, TRAIT_CLUMSY, DRUNK_TRAIT)
+				to_chat(src, span_notice("You feel sober enough to operate heavy machinery again!"))
+			else if (DT_PROB((drunkenness * 0.1) - 6, delta_time)) // getting so drunk your confidence tells you that you can totally operate a firearm without literally shooting yourself in the foot
+				to_chat(src, span_notice("You feel sober enough to operate heavy machinery again!"))
```

Comment:
there should probably be some sort of cap on how many times this message can be shown to you